Lib/unittest/case.py

```diff
@@ -84,6 +84,20 @@ def testPartExecutor(self, test_case, isTest=False):
 def _id(obj):
     return obj
 
+
+_module_cleanups = []
+def addModuleCleanup(function, *args, **kwargs):
+    """Same as addCleanup, except the cleanup items are called even if
+    setUpModule fails (unlike tearDownModule)."""
+    _module_cleanups.append((function, args, kwargs))
+
+def doModuleCleanups():
+    """Execute all module cleanup functions. Normally called for you after
+    tearDownModule."""
+    while _module_cleanups:
+        function, args, kwargs = _module_cleanups.pop()
+        function(*args, **kwargs)
```

Comment:
What happens if an exception occurs in `function`?   Currently that will prevent other module cleanups from being called.  Without looking at what's done for other cleanup functions, I think it might be better to catch the exception and continue processing, but OTOH consistency in behavior with other cleanups is best.

Reply:
Yes you're right, this doesn't work quite right here. The normal tearDown() neatly handles the exceptions so that all the cleanups will be called. I'll look into doing that here an in the class tearDown as well. Thanks for catching this!